File: leetcode/golang/119.杨辉三角II.go
func getRow(rowIndex int) []int {
    var d map[int][]int = map[int][]int{0: {1}, 1:{1, 1}};
    if _, ok := d[rowIndex];ok {
        return d[rowIndex];
    };

    tmp := getRow(rowIndex - 1);
    ret := make([]int, rowIndex + 1);
    i := 0;
    for i <= rowIndex {
        if i == 0 || i == rowIndex {
            ret[i] = 1;
        } else {
            ret[i] = tmp[i] + tmp[i - 1];
        }
        i += 1;
    }
    return ret;
}

File: leetcode/python/119.杨辉三角II.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        d = {0: [1], 1: [1, 1]}
        if rowIndex in d:
            return d[rowIndex]
        
        tmp = self.getRow(rowIndex - 1)
        ret = [1 for _ in range(rowIndex + 1)]
        for i in range(1, rowIndex):
            ret[i] = tmp[i] + tmp[i - 1]
        d[rowIndex] = ret
        return ret